src/ert/ensemble_evaluator/_ensemble.py

```diff
@@ -158,16 +157,21 @@ def _create_snapshot(self) -> Snapshot:
             metadata=self.metadata,
         )
 
-        return Snapshot(top.model_dump())
+        return Snapshot.from_nested_dict(top.model_dump())
 
     def get_successful_realizations(self) -> List[int]:
         return self.snapshot.get_successful_realizations()
 
-    def update_snapshot(self, events: List[CloudEvent]) -> PartialSnapshot:
-        snapshot_mutate_event = PartialSnapshot(self.snapshot)
+    def update_snapshot(self, events: List[CloudEvent]) -> Snapshot:
+        snapshot_mutate_event = Snapshot()
+
         for event in events:
-            snapshot_mutate_event.from_cloudevent(event)
-        self.snapshot.merge_event(snapshot_mutate_event)
+            snapshot_mutate_event = (
+                self.snapshot.update_from_cloudevent_and_generate_update_snapshot(
+                    event, snapshot_mutate_event
+                )
+            )
+        # JONAK self.snapshot.merge_event(snapshot_mutate_event)
```

Comment:
...merge_event(...) is implicitely called in the ...update_from_cloudevent...() method.